c_src/sasl_auth.c

```diff
@@ -114,6 +201,23 @@ static int sasl_cyrus_cb_getsimple(void* context, int id, const char** result, u
     return *result ? SASL_OK : SASL_FAIL;
 }
 
+static int sasl_auth_process_check(ErlNifEnv* env, sasl_state_t* state)
```

Comment:
In reviewing the code, I simply thought to myself, without a check like this in place, the library is unsafe. That is to say, if a user of this library (devoid of brod) is not aware of thread safety issues, it will blow up in their face.